dace/frontend/python/newast.py

```diff
@@ -3669,16 +3718,30 @@ def _parse_sdfg_call(self, funcname: str,
             # Handle nested closure
             closure_arrays = getattr(fcopy, '__sdfg_closure__',
                                      lambda *args: {})()
+            if hasattr(fcopy, '__sdfg_closure__'):
+                closure_arrays_only = fcopy.__sdfg_closure__(arrays_only=True)
+            else:
+                closure_arrays_only = closure_arrays
             for aname, arr in closure_arrays.items():
-                desc = data.create_datadescriptor(arr)
-                outer_name = self.sdfg.add_datadesc(aname,
-                                                    desc,
-                                                    find_new_name=True)
-                if not desc.transient:
-                    self.nested_closure_arrays[outer_name] = (arr, desc)
-                    # Add closure arrays as function arguments
+                if aname in sdfg.symbols:
```

Comment:
@alexnick83 I'm just wondering why this doesn't cover the "arrays only" case?

Reply:
I am not sure what you mean. The main issue is that closure_arrays that have to do with callbacks must not add descriptors (line 3735 and below) because they are symbols. If you are suggesting that the existence of `aname` in sdfg.symbols is enough to make the distinction, then I don't know if this could always be the case (perhaps some kind of name clashing?).

Reply:
Fixed (waiting for response from #891)